docs/wiki/EL/devp2p.md

```diff
@@ -2,6 +2,286 @@
+| **Handshake**           | None | Secure session establishment |
+| **Service Discovery**   | Limited | Topic-based lookup |
+| **Extensibility**       | Static | Supports multiple identity schemes |
+| **Clock Dependence**    | Required | Eliminated |
+| **Scalability**         | Moderate | Optimized for large networks |
+
+### 3.RLPx Transport Protocol
+
+**RLPx** is a **TCP-based encrypted transport protocol** for peer-to-peer (p2p) communication in Ethereum.It facilitates **secure message exchange** and **capability negotiation** while supporting multiple protocols over a single connection.RLPx is named after the RLP serialization format.
+
+
+
+#### ECIES Encryption
+
+- RLPx uses **Elliptic Curve Integrated Encryption Scheme (ECIES)** for secure **handshaking and session establishment**.
+- The cryptosystem consists of:
+  - **Elliptic Curve**: secp256k1
+  - **Key Derivation Function (KDF)**: NIST SP 800-56 Concatenation KDF
+  - **Message Authentication Code (MAC)**: HMAC-SHA-256
+  - **Encryption Algorithm**: AES-128-CTR
+
+##### Encryption Process
+
+1. **Initiator generates a random ephemeral keypair**.
+2. Computes **shared secret** using **Elliptic Curve Diffie-Hellman (ECDH)**.
+3. Derives encryption (`kE`) and MAC (`kM`) keys from the **shared secret**.
+4. Encrypts the message using **AES-128-CTR**.
+5. Computes a **MAC** over the encrypted message for integrity.
+6. Sends the encrypted payload.
+
+##### Decryption Process
+
+1. **Recipient extracts the sender’s ephemeral public key**.
+2. Computes the **shared secret** using **ECDH**.
+3. Derives `kE` and `kM`, then verifies the **MAC**.
+4. **Decrypts** the message using **AES-128-CTR**.
+
+
+##### Node Identity
+
+- **Ethereum nodes maintain a persistent secp256k1 keypair** for identity.
+- The **public key** serves as the **Node ID**.
+- The **private key is stored securely** and remains unchanged across sessions.
+
+
+##### Initial Handshake
+
+- RLPx connections are **established over TCP** and use an **encrypted handshake** to negotiate session keys.
+
+##### Handshake Process
+
+1. **Initiator sends an `auth` message**, encrypted using the recipient’s public key.
+2. **Recipient decrypts and verifies** the `auth` message.
+3. **Recipient responds with `auth-ack`**, containing an ephemeral public key.
+4. **Both nodes derive shared session keys** from the exchanged ephemeral keys.
+5. **The first encrypted frame** (containing the `Hello` message) is sent.
+6. **Handshake is complete** once both nodes authenticate the first frame.
+
+##### Generated Secrets
+
+| Secret | Description |
+|--------|------------|
+| `static-shared-secret` | `ECDH(node-private-key, remote-node-pubkey)` |
+| `ephemeral-key` | `ECDH(ephemeral-private-key, remote-ephemeral-pubkey)` |
+| `shared-secret` | `keccak256(ephemeral-key || keccak256(nonce || initiator-nonce))` |
+| `aes-secret` | `keccak256(ephemeral-key || shared-secret)` |
+| `mac-secret` | `keccak256(ephemeral-key || aes-secret)` |
+
+
```

Comment:
explaining the difference between static-shared-secret and ephemeral-key in terms of security (ephemeral keys provide forward secrecy) would be a good addition.